From 004963a037e04312f101bfc21910c2fc99740836 Mon Sep 17 00:00:00 2001
From: Xie XiuQi <xiexiuqi@huawei.com>
Date: Fri, 26 Feb 2021 20:20:54 +0800
Subject: [PATCH] arm64/mpam: add group partid/pmg to tasks show
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 004963a037e04312f101bfc21910c2fc99740836


hulk inclusion
category: feature
feature: ARM MPAM support
bugzilla: 48265
CVE: NA

--------------------------------

Signed-off-by: Xie XiuQi <xiexiuqi@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Signed-off-by: Wang ShaoBo <bobo.shaobowang@huawei.com>
Reviewed-by: Cheng Jian <cj.chengjian@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm64/kernel/mpam.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/kernel/mpam.c b/arch/arm64/kernel/mpam.c
index eeed3b63da4f..5883b9b2fedb 100644
--- a/arch/arm64/kernel/mpam.c
+++ b/arch/arm64/kernel/mpam.c
@@ -822,8 +822,9 @@ static void show_resctrl_tasks(struct rdtgroup *r, struct seq_file *s)
 	for_each_process_thread(p, t) {
 		if ((r->type == RDTCTRL_GROUP && t->closid == r->closid) ||
 		    (r->type == RDTMON_GROUP && t->rmid == r->mon.rmid))
-			seq_printf(s, "%d: partid = %d, pmg = %d\n",
-				   t->pid, t->closid, t->rmid);
+			seq_printf(s, "%d: partid = %d, pmg = %d, (group: partid %d, pmg %d)\n",
+				   t->pid, t->closid, t->rmid,
+				   r->closid, r->mon.rmid);
 	}
 	rcu_read_unlock();
 }
-- 
2.26.2

